pipelines/colorssn/colorssn.nf

```diff
@@ -1,65 +1,8 @@
-        path 'swissprot_singletons_desc.txt', emit: 'swissprot_singletons_desc'
-        path 'ssn-sequences.fa', emit: 'ssn_sequences'
-
-        // path 'cluster-data/uniprot-nodes', emit: 'uniprot_id_dir', optional: true
-        // path 'cluster-data/uniprot-domain-nodes', emit: 'uniprot_domain_nodes', optional: true
-        // path 'cluster-data/uniref50-nodes', emit: 'uniref50_id_dir', optional: true
-        // path 'cluster-data/uniref50-domain-nodes', emit: 'uniref50_domain_nodes', optional: true
-        // path 'cluster-data/uniref90-nodes', emit: 'uniref90_id_dir', optional: true
-        // path 'cluster-data/uniref90-domain-nodes', emit: 'uniref90_domain_nodes', optional: true
-        // path 'mapping_table_domain.txt', emit: 'mapping_table_domain', optional: true
-
-    """
-    perl $projectDir/cluster_gnn/cluster_gnn.pl -output-dir . \
-                                                             -ssnin $ssn_file \
-                                                             -ssnout ssn_out.xgmml \
-                                                             -uniprot-id-dir cluster-data/uniprot-nodes \
-                                                             -uniprot-domain-id-dir cluster-data/uniprot-domain-nodes \
-                                                             -uniref50-id-dir cluster-data/uniref50-nodes \
-                                                             -uniref50-domain-id-dir cluster-data/uniref50-domain-nodes \
-                                                             -uniref90-id-dir cluster-data/uniref90-nodes \
-                                                             -uniref90-domain-id-dir cluster-data/uniref90-domain-nodes \
-                                                             -id-out mapping_table.txt \
-                                                             -id-out-domain mapping_table_domain.txt \
-                                                             -config ${params.efi_config} \
-                                                             -efi-db ${params.efi_db} \
-                                                             -stats stats.txt \
-                                                             -conv-ratio conv_ratio.txt \
-                                                             -cluster-sizes cluster_sizes.txt \
-                                                             -cluster-num-map cluster_num_map.txt \
-                                                             -sp-clusters-desc swissprot_clusters_desc.txt \
-                                                             -sp-singletons-desc swissprot_singletons_desc.txt
-    """
-}
+include { color_and_retrieve } from "../shared/nextflow/color_workflow.nf"
 
 workflow {
-    if (params.ssn_input =~ /\.zip/) {
-        ssn_file = unzip_input(params.ssn_input)
-    } else {
-        ssn_file = params.ssn_input
-    }
```

Comment:
Would it be possible to test for the magic number (`PK\x03\x04` I think) as well as the extension? I'm just thinking about a user misnaming their file, possibly maliciously